### ISSUE TITLE
Don't keep keys in the URL

### ISSUE THREAD

**Issue by fbertsch** [mozilla/telemetry-dashboard#257]
if you switch from one histogram to another (or to a different OS for the same histogram, where the possible key values are different) you end up with an empty screen because the keys are kept from the last query and are invalid for the new query.

**Comment by ab0092** [comment]
@chutten  Is this issue occurring in the measurement-dashboard ?

**Comment by chutten** [comment]
@ab0092 

We store a lot of state in the URL so that you can share a link to the exact same dashboard you're looking at. This is important so we can collaboratively look at the same views of the data in the same time.

Now, I can't seem to reproduce this issue as stated. We need more information before we can get started. This is where you ask the reporter (@frank) how he managed to cause the issue and what it looked like.

**Comment by chutten** [comment]
And, yes, the label suggests that this was happening on the measurement dashboard

**Comment by ab0092** [comment]
@chutten then we have to wait till we get a response from @frank about how to replicate the same scenario.

**Comment by chutten** [comment]
You can start looking for the code that saves and restores the keys into and out of the URL in the meantime, and see if you can reproduce the problem by looking at keyed histograms such as IPC_SYNC_RECEIVE_MS and FORM_FILL_RESPONSE_TIME_MS

**Comment by ab0092** [comment]
Sure. I will get started right away. Thanks..

**Comment by ab0092** [comment]
@chutten i was trying out different search filters and one filters showed up results with a ??? sign, please let me know if this is an issue, try this link 
https://telemetry.mozilla.org/new-pipeline/dist.html#!cumulative=0&end_date=2017-10-01&keys=%253F%253F%253F!PPluginInstance%253A%253AMsg_ShowDirectBitmap!PPluginInstance%253A%253AMsg_ShowDirectDXGISurface!PVideoDecoderManager%253A%253AMsg_PVideoDecoderConstructor&max_channel_version=nightly%252F58&measure=IPC_SYNC_RECEIVE_MS&min_channel_version=null&os=Windows_NT%252C6.3&processType=*&product=Firefox&sanitize=1&sort_keys=submissions&start_date=2017-09-21&table=0&trim=1&use_submission_date=0

**Comment by ab0092** [comment]
There is also a design issue with the graph numbers showing below each graph bar which are overlapping one another which is making it difficult to read.

**Comment by chutten** [comment]
The key "???" is actually used in the IPC code that's reporting data. It's a valid key.

**Comment by shikhar-scs** [comment]
@chutten I've managed to find where the keys get included in and excluded out of the main url. How shall I proceed further ?

**Comment by chutten** [comment]
@shikhar-scs The idea here is to ensure that when you switch parameters (OS, Probe, etc) the keys should be removed from the URL. But we still want the keys in the URL for sharing, and to restore the key information if it was the initial state.

Does that make sense?

**Comment by shikhar-scs** [comment]
Yeah that does make sense, though, the thought of implementing the same is a bit scary.

**Comment by shikhar-scs** [comment]
I hope you'll guide me through the way :+1:

**Comment by chutten** [comment]
That's the plan 👍
